elide-datastore/elide-datastore-hibernate/src/test/java/com/yahoo/elide/datastores/hibernate/hql/AbstractHQLQueryBuilderTest.java

```diff
@@ -179,20 +190,20 @@ public void testSettingQueryParams() {
     public void testSettingQueryPagination() {
         Query query = mock(Query.class);
 
-        Optional<Pagination> previousPagination = pagination;
-
         PaginationImpl paginationMock = mock(PaginationImpl.class);
         when(paginationMock.getLimit()).thenReturn(10);
         when(paginationMock.getOffset()).thenReturn(50);
 
-        pagination = Optional.of(paginationMock);
+        EntityProjection previousEntityProjection = entityProjection;
+        entityProjection = mock(EntityProjection.class);
```

Comment:
Setting and resetting the mock this way (including the old code) seems fairly hacky.

I suggest we initialize this class with a mock entity projection.  Let's add a `@BeforeEach` or `AfterEach` annotation which resets the mocks.  

Each method will have to setup the projection the way that. makes the most sense.